bundle/permissions/workspace_root.go

```diff
@@ -52,16 +55,58 @@ func giveAccessForWorkspaceRoot(ctx context.Context, b *bundle.Bundle) error {
 	}
 
 	w := b.WorkspaceClient().Workspace
-	obj, err := w.GetStatusByPath(ctx, b.Config.Workspace.RootPath)
+	rootPath := b.Config.Workspace.RootPath
+	paths := []string{}
+	if !libraries.IsVolumesPath(rootPath) {
+		paths = append(paths, rootPath)
+	}
+
+	if !strings.HasSuffix(rootPath, "/") {
+		rootPath += "/"
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.ArtifactPath, rootPath) &&
+		!libraries.IsVolumesPath(b.Config.Workspace.ArtifactPath) {
+		paths = append(paths, b.Config.Workspace.ArtifactPath)
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.FilePath, rootPath) &&
+		!libraries.IsVolumesPath(b.Config.Workspace.FilePath) {
+		paths = append(paths, b.Config.Workspace.FilePath)
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.StatePath, rootPath) &&
+		!libraries.IsVolumesPath(b.Config.Workspace.StatePath) {
+		paths = append(paths, b.Config.Workspace.StatePath)
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.ResourcePath, rootPath) &&
+		!libraries.IsVolumesPath(b.Config.Workspace.ResourcePath) {
+		paths = append(paths, b.Config.Workspace.ResourcePath)
+	}
```

Comment:
It would be great if we can reuse this logic between the 2 PRs.

Reply:
I'd merge Validator PR first and then would reuse the code from there